File: src/screens/AddTransactionScreen.js
import React, { useState } from 'react';
import {
  View,
  Text,
  TextInput,
  StyleSheet,
  TouchableOpacity,
  ScrollView,
  Alert,
  Platform,
} from 'react-native';
import { useTranslation } from 'react-i18next';
import { Picker } from '@react-native-picker/picker';
import DateTimePicker from '@react-native-community/datetimepicker';
import { format } from 'date-fns';
import { DatabaseHelper } from '../utils/database';

export default function AddTransactionScreen({ navigation }) {
  const { t } = useTranslation();
  const [amount, setAmount] = useState('');
  const [description, setDescription] = useState('');
  const [category, setCategory] = useState('Food');
  const [type, setType] = useState('Income');
  const [date, setDate] = useState(new Date());
  const [showDatePicker, setShowDatePicker] = useState(false);

  const categories = [
    'Food',
    'Shopping',
    'Fuel',
    'Salary',
    'Subscription',
    'Grocery',
    'Personal',
    'Travel',
    'Medicine',
    'Entertainment',
    'Bills',
    'Education',
    'Investment',
    'Others',
  ];

  const getLocalizedCategories = () => {
    return categories.map((cat) => ({
      value: cat,
      label: t(`category_${cat.toLowerCase()}`),
    }));
  };

  const handleSave = async () => {
    if (!amount || !description) {
      Alert.alert(t('allFieldsRequired'));
      return;
    }

    try {
      const parsedAmount = parseFloat(amount);
      if (isNaN(parsedAmount)) {
        Alert.alert(t('invalidAmount'));
        return;
      }

      await DatabaseHelper.addTransaction(
        parsedAmount,
        category,
        type,
        date.toISOString(),
        description
      );

      Alert.alert(t('transactionSaved'));
      navigation.goBack();
    } catch (error) {
      Alert.alert(t('invalidAmount'));
    }
  };

  const onDateChange = (event, selectedDate) => {
    setShowDatePicker(false);
    if (selectedDate) {
      setDate(selectedDate);
    }
  };

  return (
    <View style={styles.container}>
      <View style={styles.header}>
        <TouchableOpacity onPress={() => navigation.goBack()}>
          <Text style={styles.backButton}>←</Text>
        </TouchableOpacity>
        <Text style={styles.headerTitle}>{t('addTransaction')}</Text>
        <View style={{ width: 30 }} />
      </View>

      <ScrollView style={styles.content}>
        <Text style={styles.label}>{t('howMuch')}</Text>
        <View style={styles.amountContainer}>
          <Text style={styles.currency}>৳</Text>
          <TextInput
            style={styles.amountInput}
            placeholder={t('enterAmount')}
            value={amount}
            onChangeText={setAmount}
            keyboardType="numeric"
          />
        </View>

        <Text style={styles.label}>{t('category')}</Text>
        <View style={styles.pickerContainer}>
          <Picker
            selectedValue={category}
            onValueChange={(itemValue) => setCategory(itemValue)}
            style={styles.picker}
          >
            {getLocalizedCategories().map((cat) => (
              <Picker.Item
                key={cat.value}
                label={cat.label}
                value={cat.value}
              />
            ))}
          </Picker>
        </View>

        <Text style={styles.label}>{t('description')}</Text>
        <TextInput
          style={styles.input}
          placeholder={t('description')}
          value={description}
          onChangeText={setDescription}
        />

        <Text style={styles.label}>{t('date')}</Text>
        <TouchableOpacity
          style={styles.dateButton}
          onPress={() => setShowDatePicker(true)}
        >
          <Text>{format(date, 'MMMM d, yyyy')}</Text>
        </TouchableOpacity>

        {showDatePicker && (
          <DateTimePicker
            value={date}
            mode="date"
            display="default"
            onChange={onDateChange}
          />
        )}

        <View style={styles.typeContainer}>
          <TouchableOpacity
            style={[
              styles.typeButton,
              type === 'Income' && styles.typeButtonActive,
              type === 'Income' && { backgroundColor: 'green' },
            ]}
            onPress={() => setType('Income')}
          >
            <Text
              style={[
                styles.typeText,
                type === 'Income' && styles.typeTextActive,
              ]}
            >
              {t('income')}
            </Text>
          </TouchableOpacity>

          <TouchableOpacity
            style={[
              styles.typeButton,
              type === 'Expense' && styles.typeButtonActive,
              type === 'Expense' && { backgroundColor: 'red' },
            ]}
            onPress={() => setType('Expense')}
          >
            <Text
              style={[
                styles.typeText,
                type === 'Expense' && styles.typeTextActive,
              ]}
            >
              {t('expense')}
            </Text>
          </TouchableOpacity>
        </View>

        <TouchableOpacity style={styles.saveButton} onPress={handleSave}>
          <Text style={styles.saveButtonText}>{t('continueButton')}</Text>
        </TouchableOpacity>
      </ScrollView>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#FDF7F0',
  },
  header: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    padding: 20,
    paddingTop: 50,
  },
  backButton: {
    fontSize: 30,
    color: '#000',
  },
  headerTitle: {
    fontSize: 18,
    fontWeight: 'bold',
    color: '#000',
  },
  content: {
    padding: 20,
  },
  label: {
    fontSize: 18,
    fontWeight: 'bold',
    marginBottom: 10,
    marginTop: 10,
  },
  amountContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    borderWidth: 1,
    borderColor: '#ccc',
    borderRadius: 8,
    paddingHorizontal: 12,
  },
  currency: {
    fontSize: 20,
    marginRight: 8,
  },
  amountInput: {
    flex: 1,
    padding: 12,
    fontSize: 16,
  },
  input: {
    borderWidth: 1,
    borderColor: '#ccc',
    borderRadius: 8,
    padding: 12,
    fontSize: 16,
  },
  pickerContainer: {
    borderWidth: 1,
    borderColor: '#ccc',
    borderRadius: 8,
    overflow: 'hidden',
  },
  picker: {
    height: 50,
  },
  dateButton: {
    borderWidth: 1,
    borderColor: '#ccc',
    borderRadius: 8,
    padding: 12,
  },
  typeContainer: {
    flexDirection: 'row',
    justifyContent: 'center',
    gap: 12,
    marginVertical: 20,
  },
  typeButton: {
    paddingHorizontal: 24,
    paddingVertical: 10,
    borderRadius: 20,
    borderWidth: 1,
    borderColor: '#ccc',
  },
  typeButtonActive: {
    borderColor: 'transparent',
  },
  typeText: {
    color: '#000',
  },
  typeTextActive: {
    color: '#fff',
    fontWeight: 'bold',
  },
  saveButton: {
    backgroundColor: '#000',
    padding: 16,
    borderRadius: 8,
    alignItems: 'center',
    marginTop: 20,
    marginBottom: 40,
  },
  saveButtonText: {
    color: '#fff',
    fontSize: 16,
    fontWeight: 'bold',
  },
});

File: lib/screens/add_transaction_screen.dart
import 'package:flutter/material.dart';
import 'package:intl/intl.dart';
import '../db/database_helper.dart';
import 'package:expensemate_rohit/generated/app_localizations.dart';

class AddTransactionScreen extends StatefulWidget {
  const AddTransactionScreen({super.key});

  @override
  State<AddTransactionScreen> createState() => _AddTransactionScreenState();
}

class _AddTransactionScreenState extends State<AddTransactionScreen> {
  final TextEditingController _amountController = TextEditingController();
  final TextEditingController _descController = TextEditingController();
  final TextEditingController _dateController = TextEditingController();

  final List<String> _categories = [
    'Food',
    'Shopping',
    'Fuel',
    'Salary',
    'Subscription',
    'Grocery',
    'Personal',
    'Travel',
    'Medicine',
    'Entertainment',
    'Bills',
    'Education',
    'Investment',
    'Others',
  ];

  String _selectedCategory = 'Food';
  String _selectedType = 'Income';
  DateTime _selectedDate = DateTime.now();

  late List<String> _localizedCategories;

  @override
  void initState() {
    super.initState();
  }

  @override
  void didChangeDependencies() {
    super.didChangeDependencies();
    _updateDateText();
    _localizedCategories =
        _getLocalizedCategories(AppLocalizations.of(context)!);
  }

  void _updateDateText() {
    final locale = Localizations.localeOf(context).toString();
    _dateController.text = DateFormat.yMMMd(locale).format(_selectedDate);
  }

  List<String> _getLocalizedCategories(AppLocalizations local) {
    return [
      local.category_food,
      local.category_shopping,
      local.category_fuel,
      local.category_salary,
      local.category_subscription,
      local.category_grocery,
      local.category_personal,
      local.category_travel,
      local.category_medicine,
      local.category_entertainment,
      local.category_bills,
      local.category_education,
      local.category_investment,
      local.category_others,
    ];
  }

  void _pickDate() async {
    final picked = await showDatePicker(
      context: context,
      initialDate: _selectedDate,
      firstDate: DateTime(2020),
      lastDate: DateTime(2100),
      locale: Localizations.localeOf(context),
    );

    if (picked != null) {
      setState(() {
        _selectedDate = picked;
        _updateDateText();
      });
    }
  }

  void _saveTransaction() async {
    final local = AppLocalizations.of(context)!;
    final amountText = _amountController.text.trim();
    final desc = _descController.text.trim();

    if (amountText.isEmpty || desc.isEmpty) {
      ScaffoldMessenger.of(context).showSnackBar(
        SnackBar(content: Text(local.allFieldsRequired)),
      );
      return;
    }

    try {
      final amount = double.parse(amountText);
      final formattedDate = _selectedDate.toIso8601String();

      await DatabaseHelper.instance.addTransaction(
        amount,
        _selectedCategory, // English name
        _selectedType,
        formattedDate,
        desc,
      );

      ScaffoldMessenger.of(context).showSnackBar(
        SnackBar(content: Text(local.transactionSaved)),
      );

      Navigator.pop(context);
    } catch (e) {
      ScaffoldMessenger.of(context).showSnackBar(
        SnackBar(content: Text(local.invalidAmount)),
      );
    }
  }

  @override
  void dispose() {
    _amountController.dispose();
    _descController.dispose();
    _dateController.dispose();
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    final local = AppLocalizations.of(context)!;

    return Scaffold(
      backgroundColor: const Color(0xFFFDF7F0),
      appBar: AppBar(
        backgroundColor: const Color(0xFFFDF7F0),
        elevation: 0,
        title: Text(
          local.addTransaction,
          style: const TextStyle(color: Colors.black),
        ),
        iconTheme: const IconThemeData(color: Colors.black),
      ),
      body: Padding(
        padding: const EdgeInsets.all(20.0),
        child: ListView(
          children: [
            Text(
              local.howMuch,
              style:
              const TextStyle(fontSize: 18, fontWeight: FontWeight.bold),
            ),
            const SizedBox(height: 10),
            TextField(
              controller: _amountController,
              keyboardType:
              const TextInputType.numberWithOptions(decimal: true),
              decoration: InputDecoration(
                prefixIcon: const Padding(
                  padding: EdgeInsets.all(12.0),
                  child: Text('৳', style: TextStyle(fontSize: 20)),
                ),
                hintText: local.enterAmount,
                border: const OutlineInputBorder(),
              ),
            ),
            const SizedBox(height: 20),

            // Localized Category Dropdown
            DropdownButtonFormField<String>(
              value: _localizedCategories[_categories.indexOf(_selectedCategory)],
              items: _localizedCategories
                  .map(
                    (label) =>
                    DropdownMenuItem(value: label, child: Text(label)),
              )
                  .toList(),
              onChanged: (val) {
                final index = _localizedCategories.indexOf(val!);
                setState(() {
                  _selectedCategory = _categories[index]; // Store English value
                });
              },
              decoration: InputDecoration(
                labelText: local.category,
                border: const OutlineInputBorder(),
              ),
            ),

            const SizedBox(height: 20),
            TextField(
              controller: _descController,
              decoration: InputDecoration(
                labelText: local.description,
                border: const OutlineInputBorder(),
              ),
            ),
            const SizedBox(height: 20),

            // Income / Expense Type Selection
            Row(
              mainAxisAlignment: MainAxisAlignment.center,
              children: [
                ChoiceChip(
                  label: Text(local.income),
                  selected: _selectedType == 'Income',
                  selectedColor: Colors.green,
                  onSelected: (_) =>
                      setState(() => _selectedType = 'Income'),
                ),
                const SizedBox(width: 12),
                ChoiceChip(
                  label: Text(local.expense),
                  selected: _selectedType == 'Expense',
                  selectedColor: Colors.red,
                  onSelected: (_) =>
                      setState(() => _selectedType = 'Expense'),
                ),
              ],
            ),
            const SizedBox(height: 20),

            TextFormField(
              controller: _dateController,
              readOnly: true,
              onTap: _pickDate,
              decoration: InputDecoration(
                labelText: local.date,
                border: const OutlineInputBorder(),
                suffixIcon: const Icon(Icons.calendar_today),
              ),
            ),

            const SizedBox(height: 30),
            ElevatedButton(
              onPressed: _saveTransaction,
              style: ElevatedButton.styleFrom(
                backgroundColor: Colors.deepPurple,
                padding: const EdgeInsets.symmetric(vertical: 16),
              ),
              child: Text(
                local.continueButton,
                style:
                const TextStyle(fontSize: 16, color: Colors.white),
              ),
            ),
          ],
        ),
      ),
    );
  }
}
